### PR TITLE
WIP Jmap vacation set

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/VacationResponseSetMethodContract.scala [hunk 1]
```diff
@@ -711,7 +707,7 @@ trait VacationResponseSetMethodContract {
          |				"notCreated": {
          |					"singleton": {
          |						"type": "invalidArguments",
-         |						"description": ""
+         |						"description": "'create' is not supported on singletong objects"
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/VacationResponseSetMethodContract.scala:710]:
```suggestion
         |						"description": "'create' is not supported on singleton objects"
```

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/VacationResponseSetMethodContract.scala [hunk 2]
```diff
@@ -767,7 +760,7 @@ trait VacationResponseSetMethodContract {
          |				"notDestroyed": {
          |					"singleton": {
          |						"type": "invalidArguments",
-         |						"description": ""
+         |						"description": "'destroy' is not supported on singletong objects"
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/VacationResponseSetMethodContract.scala:763]:
```suggestion
         |						"description": "'destroy' is not supported on singleton objects"
```

**Comment by Arsnael** [comment]
```
[36b124687dd65656753f3a0fad1dad2e953182df] [ERROR] Errors: 
[36b124687dd65656753f3a0fad1dad2e953182df] [ERROR]   ActiveMQMailQueueTest.browseShouldReturnMailsWithMimeMessage » NoSuchElement
```
not related...

test this please

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/VacationResponseSetMethod.scala [hunk 3]
```diff
@@ -77,14 +87,26 @@ class VacationResponseSetMethod @Inject() (serializer: Serializer,
...
+    SFlux.fromIterable(vacationResponseSetRequest.parsePatch()
+      .map[SMono[VacationResponseUpdateResult]]({
+        case (id, Right(patch)) =>
+          patch.asVacationPatch
+            .fold(
+              e => SMono.just(VacationResponseUpdateFailure(id, e)),
+              validatedPatch => update(validatedPatch, mailboxSession))
+        case (id, Left(e)) => SMono.just(VacationResponseUpdateFailure(id, e))
+      }))
+      .flatMap[VacationResponseUpdateResult](a =>  a)
```
Comment (LanKhuat) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/VacationResponseSetMethod.scala:100]: extra space

Comment (chibenwa) [reply server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/VacationResponseSetMethod.scala:100]: and variable name ;-)

**Comment by chibenwa** [comment]
`MemorySpamAssassinContractTest.deletingSpamMessageShouldNotImpactSpamAssassinState » ConditionTimeout`

test this please

**Comment by Arsnael** [comment]
Fix the conflicts with master

**Comment by LanKhuat** [comment]
cherry-picked in https://github.com/linagora/james-project/pull/3695. Should we close this one?

**Comment by chibenwa** [comment]
Yes!

**Pull request closed without merging**